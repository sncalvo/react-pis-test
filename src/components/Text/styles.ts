import { StyleSheet, TextStyle } from 'react-native';

interface TextStyles {
  base: TextStyle;
}

const styles = StyleSheet.create<TextStyles>({
  base: {},
});

export default styles;
