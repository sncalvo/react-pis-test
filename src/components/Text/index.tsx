import React from 'react';
import { StyleProp, Text as RNText, TextStyle } from 'react-native';

import styles from './styles';

interface TextProps {
  children: React.ReactNode;
  style?: StyleProp<TextStyle>;
  restProps?: unknown;
}

const Text = ({ children = '', style, ...restProps }: TextProps) => {
  return (
    <RNText style={[styles.base, style]} {...restProps}>
      {children}
    </RNText>
  );
};

export default Text;
