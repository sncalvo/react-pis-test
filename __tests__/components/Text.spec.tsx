import React from 'react';

import renderer from 'react-test-renderer';

import Text from '@components/Text';

describe('Text', () => {
  it('renders correctly', () => {
    const props = {
      children: 'test text',
    };
    const result = renderer.create(<Text {...props} />);
    expect(result).toMatchSnapshot();
  });
});
