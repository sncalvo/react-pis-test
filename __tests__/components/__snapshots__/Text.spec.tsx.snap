// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Text renders correctly 1`] = `
<Text
  style={
    Array [
      Object {},
      undefined,
    ]
  }
>
  test text
</Text>
`;
